pkg/controller/erspan.go

```diff
@@ -0,0 +1,358 @@
+			return false
+		}
+	}
+	spankey, err := cache.MetaNamespaceKeyFunc(span)
+	if err != nil {
+		ErspanPolicyLogger(cont.log, span).
+			Error("Could not create erspan key: ", err)
+		return false
+	}
+	cont.apicConn.ClearApicObjects(cont.aciNameForKey("span", spankey))
+	cont.erspanPolPods.DeleteSelectorObj(obj)
+	return true
+
+}
+
+func (cont *AciController) erspanSyncOpflexDev() {
+	cache.ListAll(cont.erspanIndexer, labels.Everything(),
+		func(spanObj interface{}) {
+			cont.queueErspanUpdate(spanObj.(*erspanpolicy.ErspanPolicy))
+		})
+}
+
+func (cont *AciController) queueErspanUpdate(span *erspanpolicy.ErspanPolicy) {
+	key, err := cache.MetaNamespaceKeyFunc(span)
+	if err != nil {
+		ErspanPolicyLogger(cont.log, span).
+			Error("Could not create span key:" + err.Error())
+		return
+	}
+	cont.erspanQueue.Add(key)
+}
+
+func (cont *AciController) initErspanPolPodIndex() {
+	cont.erspanPolPods = index.NewPodSelectorIndex(cont.log,
+		cont.podIndexer, cont.namespaceIndexer, cont.erspanIndexer,
+		cache.MetaNamespaceKeyFunc,
+		func(obj interface{}) []index.PodSelector {
+			span := obj.(*erspanpolicy.ErspanPolicy)
+			ls := &metav1.LabelSelector{MatchLabels: span.Spec.Selector.Labels}
+			return index.PodSelectorFromNsAndSelector(span.Spec.Selector.Namespace, ls)
+		},
+	)
+	cont.erspanPolPods.SetPodUpdateCallback(func(podkey string) {
+		podobj, exists, err := cont.podIndexer.GetByKey(podkey)
+		if exists && err == nil {
+			cont.queuePodUpdate(podobj.(*v1.Pod))
+		}
+	})
+	spanupdate := func(spankey string) {
+		spanobj, exists, err := cont.erspanIndexer.GetByKey(spankey)
+		if exists && err == nil {
+			cont.queueErspanUpdate(spanobj.(*erspanpolicy.ErspanPolicy))
+		}
+	}
+	spanhash := func(pod *v1.Pod) string {
+		return pod.Status.PodIP
+	}
+	// Callback when pods selected by erspan object change
+	cont.erspanPolPods.SetObjUpdateCallback(spanupdate)
+	cont.erspanPolPods.SetPodHashFunc(spanhash)
+}
+
+func (cont *AciController) handleErspanUpdate(span *erspanpolicy.ErspanPolicy) bool {
+
+	erspanLogger := ErspanPolicyLogger(cont.log, span)
+	spankey, err := cache.MetaNamespaceKeyFunc(span)
+	if err != nil {
+		erspanLogger.Error("Could not create erspan policy key: ", err)
+		return false
+	}
+	labelKey := cont.aciNameForKey("span", spankey)
+	cont.log.Debug("creating erspanpolicy")
+
+	// Source policies
+	srcGrp := apicapi.NewSpanVSrcGrp(labelKey)
+	srcName := labelKey + "_Src"
+	apicSlice := apicapi.ApicSlice{srcGrp}
+	if span.Spec.Source.AdminState != "" {
+		srcGrp.SetAttr("adminSt", span.Spec.Source.AdminState)
+	} else {
+		srcGrp.SetAttr("adminSt", "start")
+	}
+	src := apicapi.NewSpanVSrc(srcGrp.GetDn(), srcName)
+	srcGrp.AddChild(src)
+	if span.Spec.Source.Direction != "" {
+		src.SetAttr("dir", span.Spec.Source.Direction)
+	} else {
+		src.SetAttr("dir", "both")
+	}
+
+	// Build fvCEp for matching pods
+	cont.indexMutex.Lock()
+	podKeys := cont.erspanPolPods.GetPodForObj(spankey)
+	cont.log.Debug("podkeys are: ", podKeys)
```

Comment:
This could be pretty verbose. Can you remove? You're logging the individual items in the loop as well